Archives/VolFile.cpp

```diff
@@ -479,7 +479,7 @@ namespace Archives
 		for (temp = 0; temp < m_NumberOfIndexEntries; temp++)
 		{
 			// Make sure entry is valid
-			if (m_Index[temp].fileNameOffset == -1)
+			if (m_Index[temp].fileNameOffset == -1u)
```

Comment:
What does -1u mean here? I thought u casts a number to be unsigned. Here we are assigning 1 to be negative after saying it is unsigned?

I'm also not sure about the test in general now that I look at it. fileNameOffset is an unsigned value, so checking it for equivalence against -1 seems odd to me. I'm guessing -1 here means the largest unsigned value?

Reply:
Thanks for the rundown on -1u. I'm happy if it stays, esp if there is precedence. It would be nice to add a short comment to explain it for anyone unfamiliar. Funny that it fixes a warning on your compiler as it causes a warning on MSVC.

Warning C4146 unary minus operator applied to unsigned type, result still unsigned OP2Utility c:\ . . . op2archive\op2utility\archives\volfile.cpp 482